asreview/webapp/api/projects.py

```diff
@@ -171,102 +186,150 @@ def api_get_projects_stats():  # noqa: F401
+    # generate a project path
+    project_path = get_project_path(project_id, current_user)
+
+    project = ASReviewProject.create(
+        project_path,
+        project_id=project_id,
+        project_mode=project_mode,
+        project_name=project_name,
+        project_description=project_description,
+        project_authors=project_authors,
+    )
+
+    # create a database entry for this project
+    if current_app.config["AUTHENTICATION_ENABLED"]:
+        current_user.projects.append(
+            Project(project_id=project_id, folder=project_path.stem)
+        )
+        DB.session.commit()
 
     response = jsonify(project.config)
 
     return response, 201
 
 
-@bp.route('/projects/<project_id>/upgrade_if_old', methods=["GET"])
-@project_from_id
+@bp.route("/projects/<project_id>/upgrade_if_old", methods=["GET"])
+@asreview_login_required
+@project_from_id(current_user)
 def api_upgrade_project_if_old(project):
     """Get upgrade project if it is v0.x"""
 
     if not project.config["version"].startswith("0"):
-        response = jsonify(
-            message="Can only convert v0.x projects.")
-        response.headers.add('Access-Control-Allow-Origin', '*')
+        response = jsonify(message="Can only convert v0.x projects.")
         return response, 400
 
     # errors are handled by the InternalServerError
     upgrade_asreview_project_file(project.project_path)
 
-    response = jsonify({'success': True})
-    response.headers.add('Access-Control-Allow-Origin', '*')
+    response = jsonify({"success": True})
     return response
 
 
-@bp.route('/projects/<project_id>/info', methods=["GET"])
-@project_from_id
+@bp.route("/projects/<project_id>/info", methods=["GET"])
+@asreview_login_required
+@project_from_id(current_user)
 def api_get_project_info(project):  # noqa: F401
     """Get info on the article"""
-
     project_config = project.config
 
     # upgrade info of v0 projects
     if project_config["version"].startswith("0"):
         project_config = upgrade_project_config(project_config)
         project_config["projectNeedsUpgrade"] = True
 
+    # add user_id of owner to response if authenticated
+    if current_app.config["AUTHENTICATION_ENABLED"]:
+        db_project = Project.query.filter(
+            Project.project_id == project.config.get("id", 0)
+        ).one_or_none()
+        if db_project:
+            project_config["ownerId"] = db_project.owner_id
+
     return jsonify(project_config)
 
 
-@bp.route('/projects/<project_id>/info', methods=["PUT"])
-@project_from_id
+@bp.route("/projects/<project_id>/info", methods=["PUT"])
+@asreview_login_required
+@project_from_id(current_user)
 def api_update_project_info(project):  # noqa: F401
     """Get info on the article"""
 
     # rename the project if project name is changed
-    if request.form.get('name', None) is not None:
-        project.rename(request.form['name'])
+    if request.form.get("name", None) is not None:
```

Comment:
This could be more clear:
```[python]
new_project_name = request.form.get("name", None)
if (new_project_name is not None) and (project.config["name] != new_project_name:
   ....
```